graphql_compiler/compiler/emit_sql.py

```diff
@@ -103,8 +103,15 @@ def _find_used_columns(
             for field in filter_info.fields:
                 used_columns.setdefault(get_vertex_path(location), set()).add(field)
 
-    # Find foreign keys used
+    # Find columns used to emit edges
     for location, location_info in ir.query_metadata_table.registered_locations:
+        primary_key = None
+        primary_keys = (
+            sql_schema_info.vertex_name_to_table[location_info.type.name].alias().primary_key
+        )
+        if len(primary_keys) == 1:
+            primary_key = str(primary_keys[0].name)
```

Comment:
Is fall-through on the other cases expected / correct behavior, or is this just a hack?

Maybe add `elif len(primary_keys) > 1: raise NotImplementedError()` and perhaps raise `AssertionError` for unreachable code in the `else`? Let's be explicit about what we expect here, it's not at all obvious to me why `== 1` is a special case and we already had a couple of bugs around this code so that's making me feel extra uncomfortable with the current setup.